internal/storage/gcs/request.go

```diff
@@ -321,6 +321,23 @@ type Listing struct {
 	ContinuationToken string
 }
 
+// ListingFolders contains a set of folders by a call to ListFolders.
+type ListingFolders struct {
+	// Records for folders matching the listing criteria.
+	Folders []*Folder
+
+	// A continuation token, for fetching more results.
+	//
+	// If non-empty, this listing does not represent the full set of matching
+	// folders in the bucket. Call ListFolders again with the request's
+	// ContinuationToken field set to this value to continue where you left off.
+	//
+	// Note that there is no guarantee of atomicity of listings. Folders written
+	// and deleted concurrently with a single or multiple listing requests may or
+	// may not be returned.
+	ContinuationToken string
```

Comment:
who passes this continuation token ? is it kernel ?

Reply:
In the API results, we received a token, which we assign from the response (e.g., `folderList.ContinuationToken = folderIterator.PageInfo().Token`). This is the same process as with ListObjects.